setup.py

```diff
@@ -288,6 +315,124 @@ def readme():
+    # E265 and W503 excluded because I didn't find a way to apply changes automatically
+    _py_checker_command_line = [_py_checker, '--ignore=E265,W503']
+    _py_formatter_command_line = [_py_formatter, '--in-place', '--aggressive', '--aggressive']
+    _py_file_extensions = ['.py']
+
+    def _get_file_list(self, path, search_extentions):
+        files = []
+        exluded_directories_full_path = [os.path.join(path, excluded_dir)
+                                         for excluded_dir in self._project_directory_excluded]
+
+        # r=root, d=directories, f = files
+        for r, d, f in os.walk(path):
+            # match exclude pattern in current directory
+            found = False
+            for excluded_dir in exluded_directories_full_path:
+                if r.find(excluded_dir) >= 0:
+                    found = True
+
+            if found:
+                continue
+
+            for file in f:
+                filename, extention = os.path.splitext(file)
+                if extention in search_extentions:
+                    files.append(os.path.join(r, file))
+
+        return files
+
+    def initialize_options(self):
+        self.apply = 0
+
+    def finalize_options(self):
+        pass
+
+    def run(self):
+        root_dir = versioneer.get_root()
+        print("Project directory is: %s" % root_dir)
+
+        if self.apply:
+            self._c_formatter_command_line += ['-i']
+        else:
+            self._c_formatter_command_line += ['-output-replacements-xml']
+
+        import subprocess
+
+        bad_style_file_names = []
+
+        # C files handling
+        c_files = self._get_file_list(root_dir, self._c_file_extensions)
+        try:
+            for f in c_files:
+                command_output = subprocess.Popen(self._c_formatter_command_line + [f], stdout=subprocess.PIPE)
+                command_cout, command_cerr = command_output.communicate()
+                if not self.apply:
+                    if command_cout.find(b'<replacement ') > 0:
+                        bad_style_file_names.append(f)
+        except BaseException:
+            print("%s is not installed.\nPlease use: %s" % (self._c_formatter, self._c_formatter_install_msg))
+            exit(1)
+
+        # Python files handling
+        py_files = self._get_file_list(root_dir, self._py_file_extensions)
+        try:
+            for f in py_files:
+                if not self.apply:
+                    command_output = subprocess.Popen(
+                        self._py_checker_command_line + [f])
+                    returncode = command_output.wait()
+                    if returncode != 0:
+                        bad_style_file_names.append(f)
+                else:
+                    command_output = subprocess.Popen(
+                        self._py_formatter_command_line + [f])
+                    command_output.wait()
```

Comment:
maybe autopep8 also allows this, but yapf would allow to simplify this: just call `yapf.yapflib.yapf_api.FormatFile(f)` instead of using a pipe.

Reply:
It require a check for yapf module unavailability. If it is ok I will implement.

Reply:
Not sure I understand what you mean. In this python else branch you can just import it. If it's not available you will get a meaningful error. 
As mentioned elsewhere, installing packages in setup.py should be avoided. So we'd have the same issue with autopep8.

Reply:
> you can just import

What is no `yapf` package installed in the system? In this case the user take long, confused python error message. I think we need to provide simple, readable message about this. To do this I need to check the package availability first. it is not big code (several lines) but i think it should be.

Reply:
ok, I don't think the error message is hard to read. But you can just wrap the import in a try/except and print an error of your liking.